### PR TITLE
Add multiple class help buttons in inspector

### PR DESCRIPTION
**Pull request by ghost**
The property panel alternative to https://github.com/godotengine/godot/pull/13924 proposed by @groud.

I admit I actually like this a lot.

In most cases you will click on the top class doc button, e.g. `Node2D`. In terms of position this seems to me fairly similar to clicking one central doc button in the inspector's history button area as proposed by @djrm. 

demo

PR includes a hack/fix in ` scene/gui/tree.cpp` to allow buttons to use the custom cell background color.

### CODE REVIEW

File: editor/editor_node.cpp [hunk 1]
```diff
@@ -3175,6 +3173,13 @@ void EditorNode::_property_keyed(const String &p_keyed, const Variant &p_value,
 	AnimationPlayerEditor::singleton->get_key_editor()->insert_value_key(p_keyed, p_value, p_advance);
 }
 
+void EditorNode::_request_help(const String &p_class) {
+
+    printf("goto docs B %s\n", p_class.utf8().ptr());
```
Comment (akien-mga) [inline editor/editor_node.cpp]: Indentation issues, you should use tabs (and also remove this debug print btw).

File: editor/editor_node.cpp [hunk 2]
```diff
@@ -4596,6 +4601,7 @@ void EditorNode::_bind_methods() {
 	ClassDB::bind_method("_instance_request", &EditorNode::_instance_request);
 	ClassDB::bind_method("update_keying", &EditorNode::update_keying);
 	ClassDB::bind_method("_property_keyed", &EditorNode::_property_keyed);
+    ClassDB::bind_method("_request_help", &EditorNode::_request_help);
```
Comment (akien-mga) [inline editor/editor_node.cpp]: Indentation issue.

**Comment by poke1024** [comment]
Updated to make doc icons less opaque, as the main concern with this approach seems to be that the additional buttons add unnecessary additional visual noise.

buttons

I've also experimented with the following button, but I don't think this is intuitive:

bildschirmfoto 2017-12-05 um 18 21 04

**Comment by djrm** [comment]
i wonder how many times will you access the help from there (personally not many, i mean is nice to check it somtimes to know what a node does but you will most likely use the help along the script editor), considering that you have the documentation already visible in the properties tooltips, this seems too much. what would be nicer in my opinion is to have the brief and description in a tooltip when hovering over the class name, maybe signals documentation when hovering over them in the node panel, and so on, that would complement the current property inspector help system and in my opinion will be more useful than switching context to the crowded documentation view, which of course should be available but not as a primary action.

**Comment by poke1024** [comment]
I can only describe how I work in Godot (and I'm a noobie). When I write scripts, I usually refer a lot to the scene hierarchy and the inspector, asking myself: which nodes do I need to refer to in the script, which properties do I need, and what do they mean? Quite often I do find myself searching for something that is actually a property of a parent class, and then I indeed find myself typing in class names from the inspector into the class search.

**Comment by vnen** [comment]
In any case, you are able to navigate in the inheritance tree using the links at the top of the doc page, you shouldn't need to type the name manually into search:

godot windows opt tools 64_2017-12-06_13-22-11

**Comment by poke1024** [comment]
I know, but I don't want to. I'm writing a script, I want quick info on a base class or a property I see right before me. Here's what I can do:

(1) Use this PR. Takes 3 s. https://www.dropbox.com/s/ny0aa2c5dur7uev/many-buttons.mp4?dl=0

(2) Use keyboard search. Takes 4 s. https://www.dropbox.com/s/ropelwde9uccjdi/by-keyboard.mp4?dl=0

(3) Use one button and inheritance tree link. Takes at least 4 s and is confusing to me as it's a visual detour. https://www.dropbox.com/s/4kiourw0xomjoh3/one-button.mp4?dl=0

**Comment by djrm** [comment]
@poke1024 actually your PR only helps when you have one node, keep in mind that when you have more, clicking on them will get you to their editor (2D or 3D), so it will be way more distracting.

im not really against this, i guess it could be usefull sometimes, but in my opinion its use case is very limited to having the node you want to check as currently selected.

**Comment by poke1024** [comment]
@djrm You have a point there. It's not very consistent with what I had in mind (which is a kind of scene-tree-with-docs thing for script editing). Closing this.

**Pull request closed without merging**